FlowCrypt/Controllers/SignIn Other/EmailProviderViewController.swift

```diff
@@ -0,0 +1,528 @@
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        setupUI()
+    }
+
+    deinit {
+        NotificationCenter.default.removeObserver(self)
+    }
+}
+
+// MARK: - Parts
+extension EmailProviderViewController {
+    enum Section {
+        case account(AccountPart)
+        case imap(ServerPart)
+        case smtp(ServerPart)
+        case other(OtherSMTP)
+        case connect
+
+        static let numberOfSections: Int = 5
+
+        static func numberOfItems(for section: Int, state: State) -> Int {
+            switch (section, state) {
+            case (0, _):
+                return AccountPart.allCases.count
+            case (1, _), (2, _):
+                return ServerPart.allCases.count
+            case (3, .idle):
+                return 1
+            case (3, .extended):
+                return OtherSMTP.allCases.count
+            case (4, _):
+                return 1
+            default:
+                return 0
+            }
+        }
+
+        var section: Int {
+            switch self {
+            case .account: return 0
+            case .imap: return 1
+            case .smtp: return 2
+            case .other: return 3
+            case .connect: return 4
+            }
+        }
+
+        var indexPath: IndexPath {
+            let row: Int
+            switch self {
+            case let .account(part): row = part.rawValue
+            case let .imap(part): row = part.rawValue
+            case let .smtp(part): row = part.rawValue
+            case let .other(part): row = part.rawValue
+            case .connect: row = 0
+            }
+            return IndexPath(row: row, section: self.section)
+        }
+
+        init?(indexPath: IndexPath) {
+            switch indexPath.section {
+            case 0:
+                guard let part = AccountPart(rawValue: indexPath.row) else { return nil }
+                self = .account(part)
+            case 1:
+                guard let part = ServerPart(rawValue: indexPath.row) else { return nil }
+                self = .imap(part)
+            case 2:
+                guard let part = ServerPart(rawValue: indexPath.row) else { return nil }
+                self = .smtp(part)
+            case 3:
+                guard let part = OtherSMTP(rawValue: indexPath.row) else { return nil }
+                self = .other(part)
+            case 4:
+                self = .connect
+            default:
+                return nil
+            }
+        }
+    }
+
+    enum AccountPart: Int, CaseIterable {
+        case title, email, username, password
+    }
+
+    enum ServerPart: Int, CaseIterable {
+        case title, server, port, security
+    }
+
+    enum OtherSMTP: Int, CaseIterable {
+        case title, name, password
+    }
+
+    enum State: Equatable {
+        case idle, extended
+    }
+}
+
+// MARK: - Setup
+extension EmailProviderViewController {
```

Comment:
This is a Controller class but it seems to have a lot of rendering code. Could the rendering code be moved to a decorator class?

Reply:
The main idea is to keep direct rendering logic to be a ViewWcontroller responsibility, and decorator be responsible for providing with all information needed for rendering, like stylings like colors, fonts, titles, texts to the controller.